### PR TITLE
boards: canaan: Add initial support for canmv_k230

### PR DESCRIPTION
**Pull request by pigmoral**
Add DTS and SoC support for Canaan K230 Soc
Add board support for CanMV-K230

depends on `thead,machine-timer` driver in https://github.com/zephyrproject-rtos/zephyr/pull/69594

### CODE REVIEW

**Review by ycsin — CHANGES_REQUESTED** [review]
The code looks fine to me, but this needs a rebase (not merge), and the commits should be splitted into multiple logical commits (see [#56370](https://github.com/zephyrproject-rtos/zephyr/pull/56370/commits))

**Comment by Judehahh** [comment]
> The code looks fine to me, but this needs a rebase (not merge), and the commits should be splitted into multiple logical commits (see [#56370](https://github.com/zephyrproject-rtos/zephyr/pull/56370/commits))

Thanks, I will rebase when the dependent PR is merged and split it into multiple logical commits.

And I'm so sorry I pressed the "request review" button by mistake...

**Comment by kartben** [comment]
@Judehahh will you be coming back to this PR?

**Comment by Judehahh** [comment]
@kartben Yes, I will update once the dependency is merged.

File: soc/canaan/k230/Kconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_SERIES_CANAAN_K230
+	select RISCV
+	select RISCV_PRIVILEGED
+	select RISCV_HAS_PLIC
+	select RISCV_ISA_RV64I
+	select RISCV_ISA_EXT_G
+	select RISCV_ISA_EXT_C
+
```
Comment (nordicjm) [inline soc/canaan/k230/Kconfig:11]:
```suggestion
```

File: soc/canaan/k230/Kconfig.soc [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+config SOC_CANAAN_K230
+	bool
+	select SOC_SERIES_CANAAN_K230
+
+config SOC_SERIES
+	default "k230" if SOC_SERIES_CANAAN_K230
+
+config SOC
+	default "k230" if SOC_CANAAN_K230
```
Comment (nordicjm) [inline soc/canaan/k230/Kconfig.soc:16]: for the text here, it needs to match the symbol, if you have k230 then you need SOC_SERIES_K230/SOC_K230, if you have canaan_k280 then you need SOC_SERIES_CANAAN_K230/SOC_CANAAN_K230

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: soc/canaan/k230/Kconfig (hunk 1 above)
Comment (Judehahh) [reply soc/canaan/k230/Kconfig:11]: I'll update it when rebase

File: soc/canaan/k230/Kconfig.soc (hunk 2 above)
Comment (Judehahh) [reply soc/canaan/k230/Kconfig.soc:16]: Got it, I'll update it when rebase

**Comment by anatol** [comment]
`canmv_k230` looks like a nice board. can we have it in zephyr?

**Pull request closed without merging**